packages/renderer/vite.tests.setup.js

```diff
@@ -16,6 +16,61 @@
  * SPDX-License-Identifier: Apache-2.0
  ***********************************************************************/
 
+import path from 'node:path';
+import { readFileSync } from 'node:fs';
 import 'vitest-canvas-mock';
+import typescript from 'typescript';
+import { expect } from 'vitest';
 
 global.window.matchMedia = () => {};
+
+// read the given path and extract the method names from the Window interface
+function extractWindowMethods(filePath) {
+  // Read the content of the .d.ts file
+  const fileContent = readFileSync(filePath, 'utf-8');
+
+  // Create a TypeScript SourceFile
+  const sourceFile = typescript.createSourceFile(filePath, fileContent, typescript.ScriptTarget.Latest, true);
+
+  const methodNames = [];
+
+  // Visit each node in the AST
+  const visit = node => {
+    // Look for the Window interface
+    if (
+      typescript.isInterfaceDeclaration(node) &&
+      node.name.text === 'Window' // Target the "Window" interface
+    ) {
+      for (const member of node.members) {
+        if (typescript.isPropertySignature(member) && member.type && typescript.isFunctionTypeNode(member.type)) {
+          const name = member.name.text;
+          methodNames.push(name);
+        }
+      }
+    }
+
+    typescript.forEachChild(node, visit);
+  };
+
+  visit(sourceFile);
```

Comment:
nit: the sake of using the same visitor pattern I would call the visitor first time using
```suggestion
  typescript.forEachChild(sourceFile, visit);
```

Reply:
I think it might be less readable, 
- usually we are calling the visitor on the tree (first node) and it's going recursive from there
- but with your suggestion, we iterate on all the children first and do the same thing than the last part of the visitor, so I'm not even sure we're optimizing a lot.

What other people thoughts here ? I will do whatever is the most upvoted

cc @feloy @axel7083 the other reviewers

Reply:
I'm fine with the current implementation, with the first call using the `visit` function, and the call to `typescript.forEachChild` hidden into the implementation (the recursivity being an implementation detail)

Reply:
I am okey with both